tests/docker_test.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# This script logs into the Datalab service, runs a specified SQL query asynchronously,
+# waits for the query to complete, and then checks if the resulting CSV file contains
+# the expected number of lines (excluding the header).
+
+# Arguments:
+#   $1 (USER_NAME): Username for Datalab login. Mandatory.
+#   $2 (PASSWD): Password for Datalab login. If not provided, the script will asks you
+#                for one.
+
+# Usage:
+#   ./script_name.sh <USER_NAME> [<PASSWD>]
+
+# The script performs the following operations:
+#   1. Logs into Datalab with the provided username and optional password.
+#   2. Executes a predefined SQL query to fetch data from the GAIA DR3 source,
+#      limiting the results to a specified number of lines.
+#   3. Waits for the query to complete by periodically checking the job status.
+#   4. Saves the query results to a CSV file named after the job ID.
+#   5. Checks if the CSV file contains the expected number of lines (excluding the header)
+#      and prints a message indicating whether the check was successful.
+
+set -e
+
+USER_NAME=$1
+PASSWD=$2
+
+if [ -z "${PASSWD}" ]
+then
+  datalab login --user="${USER_NAME}"
+else
+  datalab login --user="${USER_NAME}" --password="${PASSWD}"
+fi
+
+# The expected number of lines
+expected_lines=100000
+export SQL="SELECT ra, dec, parallax, pmra, pmdec, radial_velocity
+FROM gaia_dr3.gaia_source
+WHERE 't' = Q3C_RADIAL_QUERY(ra, dec, 16.95, -4.1, 21.48)
+LIMIT ${expected_lines}
+"
+
+JOBID=$(datalab query --sql="$SQL" --async=True)
+echo "Job ID for submitted async SQL: $JOBID"
+
+STATUS=$(datalab qstatus --jobId="$JOBID")
+while [[ "$STATUS" != "COMPLETED" ]]
```

Comment:
This might loop forever if the phase is `ERROR` or `ABORTED`

Reply:
i.e. needs a global timeout. Should be the same as the backend timeout probably (whose value I don't recall; 6 hours? 24 hours? @mjfitzpatrick ?

Reply:
I don't think a global timeout is needed for this unless I am missing something? My comment above was about refactoring this test script a bit.

Reply:
good catch!

Reply:
Sorry, my comment was regarding the `while [[ "$STATUS" != "COMPLETED" ]]` ... If for some reason the status never reaches COMPLETED, this should still time out some time.

Reply:
> Sorry, my comment was regarding the `while [[ "$STATUS" != "COMPLETED" ]]` ... If for some reason the status never reaches COMPLETED, this should still time out some time.

Ahh ok! I I'll defer to @iglesu to consider if also having a timeout or max iterations is appropriate. I think we can reliably check for all of the phases that represent that the job is done (with or without error) which is what my comment was about but theoretically it could also get stuck in a `QUEUED` or `EXECUTING` phase which I think is what Robert was getting at. Maybe not an edge case we need to handle in this context but also it probably wouldn't be terribly difficult to implement into the existing logic.

Reply:
> Sorry, my comment was regarding the `while [[ "$STATUS" != "COMPLETED" ]]` ... If for some reason the status never reaches COMPLETED, this should still time out some time.

This issue has been fixed in this commit: 64e50868bccd6a0d9a620497f5af7c3b3a9f8563